dozer-core/src/epoch/manager.rs

```diff
@@ -33,9 +34,9 @@ enum EpochManagerStateKind {
         /// Whether sources should terminate.
         terminating: bool,
         /// - `None`: Should not commit.
-        /// - `Some(None)`: Should commit but should not persist.
-        /// - `Some(Some(index))`: Should commit and persist records from `index`.
-        next_record_index_to_persist_if_committing: Option<Option<usize>>,
+        /// - `Some(false)`: Should commit but should not persist.
+        /// - `Some(true)`: Should commit and persist records from `index`.
+        should_persist_if_committing: Option<bool>,
```

Comment:
Something like this should work
```rust
let common_info = commit_action.should_commit().then(|| EpochCommonInfo {
        id: *epoch_id,
        checkpoint_writer: commit_action.should_persist().then(
            || Arc::new(CheckpointWriter::new(self.checkpoint_factory.clone()))
        ),
    }),
}

impl CommitAction {
    fn should_persist(&self) -> bool {
        matches!(self, &CommitAndPersist)
    }
    fn should_commit(&self) -> bool {
        matches!(self, &CommitAndPersist | &Commit)
    }
}
```

Reply:
I understand it's just that these extra lines don't worth it for a private type.

Reply:
It would save me quite a few brain cycles I would otherwise spent loading the meaning of the field into my brain's registers while reading and those registers are also sparse.

A compromise might be to just split it into two bools: `commit` and `persist_if_committing`. That saves me almost the same amount of cycles while reading, without introducing an extra type.

Reply:
Haha, let me make the change now that you insist

Reply:
Thanks :)

Reply:
`bool::then` is handy. Didn't know the method before.